manager/src/main/java/org/openremote/manager/notification/EmailNotificationHandler.java

```diff
@@ -85,32 +92,59 @@ public void init(Container container) throws Exception {
 
         defaultFrom = container.getConfig().getOrDefault(OR_EMAIL_FROM, OR_EMAIL_FROM_DEFAULT);
 
-        if (!TextUtil.isNullOrEmpty(host) && !TextUtil.isNullOrEmpty(user) && !TextUtil.isNullOrEmpty(password)) {
+        if (!TextUtil.isNullOrEmpty(host) && (useOAuth || (!TextUtil.isNullOrEmpty(user) && !TextUtil.isNullOrEmpty(password)))) {
+
+            // Init client builder
             boolean startTls = getBoolean(container.getConfig(), OR_EMAIL_TLS, OR_EMAIL_TLS_DEFAULT);
             String protocol = startTls ? "smtp" : getString(container.getConfig(), OR_EMAIL_PROTOCOL, OR_EMAIL_PROTOCOL_DEFAULT);
-            Properties props = new Properties();
-            props.put("mail." + protocol + ".auth", true);
-            props.put("mail." + protocol + ".starttls.enable", startTls);
-            props.put("mail." + protocol + ".host", host);
-            props.put("mail." + protocol + ".port", port);
+            MailClientBuilder mailClientBuilder = new MailClientBuilder(
+                    container.getExecutorService(), protocol, host, port
+            );
+
+            // Add authentication
+            if(useOAuth) {
+                String oAuthUrl = container.getConfig().getOrDefault(OR_EMAIL_OAUTH2_URL, null);
+                String oAuthScopes = container.getConfig().getOrDefault(OR_EMAIL_OAUTH2_SCOPES, "");
+                if(!TextUtil.isNullOrEmpty(oAuthUrl)) {
```

Comment:
Should also check clientId and secret are provided

Reply:
It's not an amazing solution, but the `useOAuth` variable is linked to whether both client_id and client secret are set.
I wanted to prevent a new `OAUTH_ENABLE` environment variable, so I simply check whether both exist,
as stated on line 80 of this file.

Reply:
Decided to add an additional check anyway. 👍

Reply:
Sorry I missed that